wasm/src/tests.rs

```diff
@@ -248,6 +248,36 @@ fn test_run_error_program_multiple_shots() {
     .assert_eq(&output.join("\n"));
 }
 
+#[test]
+fn test_run_error_program_multiple_shots_qubit_leak() {
+    // If qubits are leaked from execution, the runtime will fail with an out of memory
+    // error pretty quickly.
```

Comment:
I'm a bit hesitant about this "pretty quickly" statement. What if it hangs for a long time instead trying to come up with a memory? Maybe we can do something more deterministic...

Reply:
out of the 100 shots, I was seeing it fail around shot 24. Leaks are hard to be deterministic about... I suppose we could try peeking into the simulator state between shots to confirm it is the same?

Reply:
I don't think I want to change how the test is invoked... the critical part of the change is that the undelrying API does the right thing with state between shot loops, and doing something like checking the state would require interrupting the loop to do other things in between. I'm going to leave this as-is, since it reliably reproduced the issue for us so far.